merge-v1/src/mapping.ts

```diff
@@ -1,74 +1,268 @@
+
+  nft.save();
+}
+
+/**
+ * @notice
+ * @param event MassUpdate
+ * @dev TODO: fix decodeValues()
+ */
+export function handleMassUpdate(event: MassUpdate): void {
+  let tokenIdBurned = event.params.tokenIdBurned.toString();
+  let tokenIdPersist = event.params.tokenIdPersist.toString();
+  let combinedMass = event.params.mass.toString();
+
+  let nftBurned = NFT.load(tokenIdBurned) as NFT;
+  let nftPersist = NFT.load(tokenIdPersist) as NFT;
+  let updatedValue = nftBurned.massSize.plus(nftPersist.massValue);
+
+  nftPersist.massValue = updatedValue;
+  nftBurned.mergedInto = tokenIdPersist;
 
-  let user = User.load(event.transaction.to.toHex()); // from my understanding, if there is no User entity for this address, then we continue through the function logic.
+  let _absorbedNFTs = nftPersist.absorbedNFTs;
+  _absorbedNFTs.push(tokenIdBurned);
+  nftPersist.absorbedNFTs = _absorbedNFTs;
 
-  let from = event.transaction.from.toHex();
-  // first, if `null` checks create new user entity if `from` was address(0)
-  // TODO: look at ENS subgraph and study how they have implemented ethereum aspects such as EMPTYADDRESS within their mapping files as per: https://github.com/ensdomains/ens-subgraph/blob/master/src/utils.ts
+  let _timestampsPersist = nftPersist.timeStamp;
+  _timestampsPersist.push(event.block.timestamp);
+  nftPersist.timeStamp = _timestampsPersist;
 
-  if (!user && from == Address(0)) {
+  // mergeCount, massSize, color, mergeClass of nftPersist
+  nftPersist.mergeCount++;
+  nftPersist.massSize = (updatedValue % CLASS_MULTIPLIER) as BigInt;
+  let _mergeClass: BigInt = updatedValue / CLASS_MULTIPLIER;
+  let mergeClass: string = checkMergeClass(_mergeClass);
+  nftPersist.mergeClass = mergeClass;
+  nftPersist.color = checkColor(mergeClass);
+
+  nftBurned.save();
+  nftPersist.save();
+
+  // TODO: when decodeValue() works, may have to tweak as I haven't looked at it in a bit.
+
+  // nftPersist.massSize = decodeValue(
+  //   updatedValue
+  // ).massSize;
+  // nftPersist.color = decodeValue(updatedValue).color;
+  // nftPersist.mergeClass = decodeValue(updatedValue).mergeClass;
+
+  // TODO: FYI: only other time MassUpdate happens is when _burnNoEmitTransfer happens, I realize now that I need to make sure this accounts for that, but I do not think I have seen the function called tbh in etherscan / don't understand when it would get called.
+}
+
+/* ========== GENERAL FUNCTIONS ========== */
+
+/**
+ * @notice load NFT or call createNFT() for new NFT
+ * @param event: Transfer from handleTransfer() caller
+ * @returns NFT with updated owner, or newly minted NFT entity
+ */
+export function getNFT(event: Transfer, user: User): NFT {
+  let tokenId = event.params.tokenId.toString();
+  let to = event.params.to.toHex();
+
+  let nft = NFT.load(tokenId);
+  if (!nft) {
+    nft = createNFT(event, user);
+    return nft;
+  } else if (to == EMPTY_ADDRESS) {
+    nft.merged = true;
+    nft.owner = EMPTY_ADDRESS;
```

Comment:
I think its better to set this to null

Reply:
Since I cleaned up the if statements that sort out creating User and NFT entities, I ended up removing the assignment of a default value anyways here. I have changed the schema though to not have a required `owner` field for `NFT` entities.